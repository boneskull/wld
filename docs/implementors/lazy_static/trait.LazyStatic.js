(function() {var implementors = {};
implementors["wld"] = [{"text":"impl LazyStatic for BLOCKTYPE_COLOR_MAP","synthetic":false,"types":[]},{"text":"impl LazyStatic for WALLTYPE_COLOR_MAP","synthetic":false,"types":[]},{"text":"impl LazyStatic for PAINT_COLOR_MAP","synthetic":false,"types":[]}];
if (window.register_implementors) {window.register_implementors(implementors);} else {window.pending_implementors = implementors;}})()