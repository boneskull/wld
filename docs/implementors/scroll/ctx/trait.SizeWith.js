(function() {var implementors = {};
implementors["wld"] = [{"text":"impl SizeWith&lt;Endian&gt; for AnglerQuestFish","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for BlockType","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for EntityType","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for EvilType","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for InvasionType","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for ItemType","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for TBool","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;TileEntityType&gt; for TileEntityType","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for WallType","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Block&gt; for Block","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for Position","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;TString&gt; for TString","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;VariableTBitVec&gt; for VariableTBitVec","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;TBitVec&gt; for TBitVec","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; SizeWith&lt;WorldCtx&lt;'a&gt;&gt; for Footer","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for HardmodeOre","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for Offsets","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for Header","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for House","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;HouseVec&gt; for HouseVec","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;ItemStack&gt; for ItemStack","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Chest&gt; for Chest","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;ItemStackVec&gt; for ItemStackVec","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Chests&gt; for Chests","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for Mob","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;MobVec&gt; for MobVec","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;NPC&gt; for NPC","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;NPCVec&gt; for NPCVec","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;PressurePlates&gt; for PressurePlates","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for Bounds","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for QuadrantStyle","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for WorldStyle","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;GeneratorInfo&gt; for GeneratorInfo","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for TUuid","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Properties&gt; for Properties","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Sign&gt; for Sign","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Signs&gt; for Signs","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for BossesSlain","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for BossesSlain2","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for PillarStatus","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for SavedNPCs","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for EventsCompleted","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for ShadowOrbStatus","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for RainStatus","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for InvasionStatus","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for Backgrounds","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for Clouds","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;AnglerQuestStatus&gt; for AnglerQuestStatus","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;MobKills&gt; for MobKills","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;PartyingNPCs&gt; for PartyingNPCs","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;PartyStatus&gt; for PartyStatus","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for SandstormStatus","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for OldOnesArmyStatus","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Status&gt; for Status","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for LogicSensor","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;TileEntity&gt; for TileEntity","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;TileEntities&gt; for TileEntities","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Liquid&gt; for Liquid","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;RunLength&gt; for RunLength","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for TileHeader","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for TileAttributes","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Endian&gt; for ExtendedTileAttributes","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Tile&gt; for Tile","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;TileVec&gt; for TileVec","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;TileMatrix&gt; for TileMatrix","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;Wall&gt; for Wall","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;World&gt; for World","synthetic":false,"types":[]},{"text":"impl SizeWith&lt;TownManager&gt; for TownManager","synthetic":false,"types":[]}];
if (window.register_implementors) {window.register_implementors(implementors);} else {window.pending_implementors = implementors;}})()