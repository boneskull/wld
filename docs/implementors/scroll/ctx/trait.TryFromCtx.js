(function() {var implementors = {};
implementors["wld"] = [{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for AnglerQuestFish","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for EntityType","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for EvilType","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for InvasionType","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for ItemType","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for TBool","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, u8, [u8]&gt; for TileEntityType","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, BlockCtx&lt;'a&gt;, [u8]&gt; for Block","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for Position <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;Position: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for TString","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for VariableTBitVec","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for TBitVec","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, WorldCtx&lt;'a&gt;, [u8]&gt; for Footer","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for HardmodeOre","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for Offsets <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;Offsets: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for Header","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for House <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;House: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for HouseVec","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for ItemStack","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, i16, [u8]&gt; for Chest","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, i16, [u8]&gt; for ItemStackVec","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for Chests","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for Mob <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;Mob: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for MobVec","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for NPC","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for NPCVec","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for PressurePlates","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for Bounds <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;Bounds: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for QuadrantStyle <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;QuadrantStyle: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for WorldStyle <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;WorldStyle: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for GeneratorInfo","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for TUuid","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for Properties <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;Properties: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for Sign <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;Sign: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for Signs","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for BossesSlain <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;BossesSlain: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for BossesSlain2 <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;BossesSlain2: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for PillarStatus <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;PillarStatus: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for SavedNPCs <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;SavedNPCs: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for EventsCompleted <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;EventsCompleted: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for ShadowOrbStatus <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;ShadowOrbStatus: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for RainStatus <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;RainStatus: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for InvasionStatus <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;InvasionStatus: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for Backgrounds <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;Backgrounds: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for Clouds <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;Clouds: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for AnglerQuestStatus","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for MobKills","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for PartyingNPCs","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for PartyStatus <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;PartyStatus: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for SandstormStatus <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;SandstormStatus: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for OldOnesArmyStatus <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;OldOnesArmyStatus: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for Status <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;Status: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for LogicSensor <span class=\"where fmt-newline\">where<br>&nbsp;&nbsp;&nbsp;&nbsp;LogicSensor: 'a,&nbsp;</span>","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for TileEntity","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for TileEntities","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, LiquidType, [u8]&gt; for Liquid","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, RLEType, [u8]&gt; for RunLength","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for TileHeader","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, Endian, [u8]&gt; for TileAttributes","synthetic":false,"types":[]},{"text":"impl&lt;'a, '_&gt; TryFromCtx&lt;'a, &amp;'_ TBitVec, [u8]&gt; for ExtendedTileAttributes","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, TileCtx&lt;'a&gt;, [u8]&gt; for Tile","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, WorldCtx&lt;'a&gt;, [u8]&gt; for TileMatrix","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, bool, [u8]&gt; for Wall","synthetic":false,"types":[]},{"text":"impl&lt;'a&gt; TryFromCtx&lt;'a, usize, [u8]&gt; for TownManager","synthetic":false,"types":[]}];
if (window.register_implementors) {window.register_implementors(implementors);} else {window.pending_implementors = implementors;}})()