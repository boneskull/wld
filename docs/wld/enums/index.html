<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="API documentation for the Rust `enums` mod in crate `wld`."><meta name="keywords" content="rust, rustlang, rust-lang, enums"><title>wld::enums - Rust</title><link rel="stylesheet" type="text/css" href="../../normalize.css"><link rel="stylesheet" type="text/css" href="../../rustdoc.css" id="mainThemeStyle"><link rel="stylesheet" type="text/css" href="../../light.css"  id="themeStyle"><link rel="stylesheet" type="text/css" href="../../dark.css" disabled ><link rel="stylesheet" type="text/css" href="../../ayu.css" disabled ><script src="../../storage.js"></script><noscript><link rel="stylesheet" href="../../noscript.css"></noscript><link rel="icon" type="image/svg+xml" href="../../favicon.svg">
<link rel="alternate icon" type="image/png" href="../../favicon-16x16.png">
<link rel="alternate icon" type="image/png" href="../../favicon-32x32.png"><style type="text/css">#crate-search{background-image:url("../../down-arrow.svg");}</style></head><body class="rustdoc mod"><!--[if lte IE 8]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="sidebar"><div class="sidebar-menu">&#9776;</div><a href='../../wld/index.html'><div class='logo-container rust-logo'><img src='../../rust-logo.png' alt='logo'></div></a><p class="location">Module enums</p><div class="sidebar-elems"><div class="block items"><ul><li><a href="#enums">Enums</a></li></ul></div><p class="location"><a href="../index.html">wld</a></p><script>window.sidebarCurrent = {name: "enums", ty: "mod", relpath: "../"};</script><script defer src="../sidebar-items.js"></script></div></nav><div class="theme-picker"><button id="theme-picker" aria-label="Pick another theme!"><img src="../../brush.svg" width="18" alt="Pick another theme!"></button><div id="theme-choices"></div></div><script src="../../theme.js"></script><nav class="sub"><form class="search-form"><div class="search-container"><div><select id="crate-search"><option value="All crates">All crates</option></select><input class="search-input" name="search" disabled autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"></div><span class="help-button">?</span>
                <a id="settings-menu" href="../../settings.html"><img src="../../wheel.svg" width="18" alt="Change settings"></a></div></form></nav><section id="main" class="content"><h1 class="fqn"><span class="out-of-band"><span id="render-detail"><a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">[<span class="inner">&#x2212;</span>]</a></span><a class="srclink" href="../../src/wld/enums.rs.html#1-26" title="goto source code">[src]</a></span><span class="in-band">Module <a href="../index.html">wld</a>::<wbr><a class="mod" href="">enums</a></span></h1><h2 id="enums" class="section-header"><a href="#enums">Enums</a></h2>
<table><tr class="module-item"><td><a class="enum" href="enum.AnglerQuestFish.html" title="wld::enums::AnglerQuestFish enum">AnglerQuestFish</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.BlockShape.html" title="wld::enums::BlockShape enum">BlockShape</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.BlockType.html" title="wld::enums::BlockType enum">BlockType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.EntityType.html" title="wld::enums::EntityType enum">EntityType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.EvilType.html" title="wld::enums::EvilType enum">EvilType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.HardmodeOreType.html" title="wld::enums::HardmodeOreType enum">HardmodeOreType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.InvasionType.html" title="wld::enums::InvasionType enum">InvasionType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.ItemType.html" title="wld::enums::ItemType enum">ItemType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.LiquidType.html" title="wld::enums::LiquidType enum">LiquidType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.RLEType.html" title="wld::enums::RLEType enum">RLEType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.TBool.html" title="wld::enums::TBool enum">TBool</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.TileEntityType.html" title="wld::enums::TileEntityType enum">TileEntityType</a></td><td class="docblock-short"></td></tr><tr class="module-item"><td><a class="enum" href="enum.WallType.html" title="wld::enums::WallType enum">WallType</a></td><td class="docblock-short"><p>A list of vanilla Walls by internal name, mapped to their IDs.</p>
</td></tr></table></section><section id="search" class="content hidden"></section><section class="footer"></section><script>window.rootPath = "../../";window.currentCrate = "wld";</script><script src="../../main.js"></script><script defer src="../../search-index.js"></script></body></html>