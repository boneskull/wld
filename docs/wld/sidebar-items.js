initSidebarItems({"fn":[["enable_logger","Enable logging for this crate given [`log::LevelFilter`] `level`."],["parse_world","Instantiate a [`World`] from a slice of bytes."]],"mod":[["constants",""],["enums",""],["models",""]]});