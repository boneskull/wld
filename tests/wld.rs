use insta::assert_debug_snapshot;
use lazy_static::lazy_static;
use scroll::ctx::SizeWith;
use std::fs::read;
use wld::{
  model::World,
  parse_world,
};

lazy_static! {
  // paths are relative to root, I guess?
  pub static ref WORLD: std::vec::Vec<u8> =
    read("tests/fixtures/Foon.wld").expect("Unable to read file");
  pub static ref PARSED_WORLD: World = parse_world(&WORLD).unwrap();
}

#[test]
fn test_parse_status_header() {
  assert_debug_snapshot!(PARSED_WORLD.status.header);
}

#[test]
fn test_parse_status_properties() {
  assert_debug_snapshot!(PARSED_WORLD.status.properties);
}

#[test]
fn test_parse_status_status() {
  assert_debug_snapshot!(PARSED_WORLD.status.status);
}

#[test]
fn test_parse_tiles() {
  // if we did the whole thing, the snapshot would be about 2GB
  assert_debug_snapshot!(&PARSED_WORLD.tiles[0][0..50]);
}

#[test]
fn test_parse_npcs() {
  assert_debug_snapshot!(PARSED_WORLD.npcs);
}

#[test]
fn test_parse_mobs() {
  assert_debug_snapshot!(PARSED_WORLD.mobs);
}

#[test]
fn test_parse_houses() {
  assert_debug_snapshot!(PARSED_WORLD.houses);
}

#[test]
fn test_parse_footer() {
  assert_debug_snapshot!(PARSED_WORLD.footer);
}

#[test]
fn test_write() {
  // wld::init_logger();
  let res = PARSED_WORLD.write();
  assert_eq!(res.unwrap().len(), World::size_with(&PARSED_WORLD));
}

// #[test]
// fn test_render() {
//   assert_eq!(PARSED_WORLD.render().dimensions(), (4200, 1200));
// }
