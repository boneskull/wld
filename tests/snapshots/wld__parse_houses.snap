---
source: tests/wld.rs
expression: PARSED_WORLD.houses
---
HouseVec(
    [
        House {
            entity_type: Guide,
            position: Position {
                x: 1932,
                y: 388,
            },
        },
        House {
            entity_type: Dd2bartender,
            position: Position {
                x: 2159,
                y: 230,
            },
        },
    ],
)
